### PR TITLE
Manually update to idl@3.35.0

### PR DESCRIPTION
**Pull request by saschanaz**

### CODE REVIEW

**Comment by saschanaz** [comment]
Oh no, SharedArrayBuffer is ES2017 thing.

File: inputfiles/overridingTypes.jsonc [hunk 1]
```diff
@@ -3553,6 +3553,11 @@
                     }
                 ],
                 "overrideType": "ReadableStreamDefaultReader<T> | ReadableStreamBYOBReader"
+            },
+            {
+                "name": "AllowSharedBufferSource",
+                // No SharedArrayBuffer in ES5, oh no.
```
Comment (saschanaz) [inline inputfiles/overridingTypes.jsonc:3559]: For now I'm just removing SharedArrayBuffer, as I don't have any better solution. cc @sandersn

**Comment by saschanaz** [comment]
lgtm

File: inputfiles/overridingTypes.jsonc (hunk 1 above)
Comment (sandersn) [reply inputfiles/overridingTypes.jsonc:3579]: Uhhh, is the problem here that DOM types have to work with any ES target, even ones that are incredibly old, like ES5/ES2015?

If so, this is going to become more of a problem over time. Maybe we could split the DOM types into parts which require newer ES types?

Comment (saschanaz) [reply inputfiles/overridingTypes.jsonc:3579]: Theoretically we can have `dom.sharedbuffer` like we have `dom.iterable`, yes. That should work for now, until we get any case that function overloads don't work (which is already a problem for `dom.iterable`). Let me file an issue for that...

Comment (saschanaz) [reply inputfiles/overridingTypes.jsonc:3579]: Filed #1579.

**Pull request merged**